pkg/server/sandbox_run.go

```diff
@@ -225,7 +228,20 @@ func (c *criContainerdService) generateSandboxContainerSpec(id string, config *r
 	// Set hostname.
 	g.SetHostname(config.GetHostname())
 
-	// TODO(random-liu): [P0] Set DNS options. Maintain a resolv.conf for the sandbox.
+	// Set DNS options.
+	if dnsConfig := config.GetDnsConfig(); dnsConfig != nil {
+		// resolvPath is not "" guaranteed by 'dnsConfig != nil'
+		resolvPath := getResolvPath(getSandboxRootDir(c.rootDir, id), config)
+		err := parseDNSOptions(dnsConfig.Servers, dnsConfig.Searches, dnsConfig.Options, resolvPath)
+		if err != nil {
+			err1 := removeFile(resolvPath)
```

Comment:
I found that we've already had code cleaning up the sandbox root. https://github.com/kubernetes-incubator/cri-containerd/blob/master/pkg/server/sandbox_run.go#L112

We can rely on that to cleanup.